Arrangement_on_surface_2/include/CGAL/Arr_point_location/Arr_trapezoid_ric_pl_impl.h

```diff
@@ -316,48 +316,41 @@ _vertical_ray_shoot(const Point_2& p, bool shoot_up) const
 // face) we check the isolated vertices inside the face to check whether there
 // is an isolated vertex right above/below the query point.
 //
-template <class Arrangement>
+template <typename Arrangement>
 typename Arr_trapezoid_ric_point_location<Arrangement>::result_type
 Arr_trapezoid_ric_point_location<Arrangement>::
 _check_isolated_for_vertical_ray_shoot (Halfedge_const_handle halfedge_found,
                                         const Point_2& p,
                                         bool shoot_up,
                                         const Td_map_item& tr) const
 {
+  const auto* gt = this->arrangement()->geometry_traits();
   const Comparison_result point_above_under = (shoot_up ? SMALLER : LARGER);
-  typename Geometry_traits_2::Compare_x_2          compare_x =
-    this->arrangement()->traits()->compare_x_2_object();
-  typename Geometry_traits_2::Compare_xy_2         compare_xy =
-    this->arrangement()->traits()->compare_xy_2_object();
-  typename Geometry_traits_2::Compare_y_at_x_2     compare_y_at_x =
-    this->arrangement()->traits()->compare_y_at_x_2_object();
-
-  Isolated_vertex_const_iterator   iso_verts_it;
-  Vertex_const_handle              closest_iso_v;
-  const Vertex_const_handle        invalid_v;
-  const Halfedge_const_handle      invalid_he;
-  Face_const_handle                face;
+  auto compare_x = gt->compare_x_2_object();
+  auto compare_xy = gt->compare_xy_2_object();
+  auto compare_y_at_x = gt->compare_y_at_x_2_object();
+
+  Vertex_const_handle closest_iso_v;
+  const Vertex_const_handle invalid_v;
+  const Halfedge_const_handle invalid_he;
 
   // If the closest feature is a valid halfedge, take its incident face.
   // Otherwise, take the unbounded face.
-  if (halfedge_found == invalid_he)
-    face = _get_unbounded_face(tr, p, All_sides_oblivious_category());
-  else
+  Face_const_handle face = (halfedge_found == invalid_he) ?
```

Comment:
This triggers a new warning:
`Arrangement_on_surface_2/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz:/mnt/testsuite/include/CGAL/Arr_point_location/Arr_trapezoid_ric_pl_impl.h:339:21: warning: operation on 'face' may be undefined [-Wsequence-point]
`